Added information about how empty voice command strings are handled (#448)
smartdevicelink/sdl_app_library_guides: docs/Speech and Audio/Setting Up Voice Commands/index.md

```
!@

### Unsupported Voice Commands

> The library automatically filters out empty strings and whitespace-only strings from a voice command's array of strings. For example, if a voice command has the following array values: `[" ", "First", "", "Voice Command"]` the library will filter it to: `["First", "Voice Command"]`.

Reviewer: ```suggestion
The library automatically filters out empty strings and whitespace-only strings from a voice command's @![iOS, javascript]array!@ @![android, javaSE, javaEE]list!@ of strings. For example, if a voice command has the following @![iOS, javascript]array!@ @![android, javaSE, javaEE]list!@ values: `[" ", "First", "", "Voice Command"]` the library will filter it to: `["First", "Voice Command"]`.
```

> A voice command that contains empty strings (or strings only containing whitespace) in its array will have those empty strings filtered out by the library. For example, if a voice command has the following array values: `[" ", "First", "", "Voice Command"]` will become `["First", "Voice Command"]`.

Reviewer 2: ```suggestion
The library automatically filters out empty strings and whitespace-only strings from a voice command's array of strings. For example, if a voice command has the following array values: `[" ", "First", "", "Voice Command"]` the library will filter it to: `["First", "Voice Command"]`.
```

> If all the voice commands contain only empty strings, the upload request will be aborted and the previous voice commands will remain available.

Reviewer 2: ```suggestion
If you provide an array of voice commands which only contains empty string and whitespace-only strings across all of the voice commands, the upload request will be aborted and the previous voice commands will remain available.
```

> If you provide an array of voice commands which only contains empty string and whitespace-only strings across all of the voice commands, the upload request will be aborted and the previous voice commands will remain available.

Reviewer: ```suggestion
If you provide a@![iOS, javascript]n array!@ @![android, javaSE, javaEE]list!@ of voice commands which only contains empty string and whitespace-only strings across all of the voice commands, the upload request will be aborted and the previous voice commands will remain available.
```

```
!@

### Unsupported Voice Commands

> Voice commands that contains empty strings in their array will be filtered out by the library since they are unsupported by the module. For example in a case where a voice command has the following array values ["\t", "First", " ", "Voice Command"] will become ["First", "Voice Command"].

Reviewer 2: ```suggestion
A voice command that contains empty strings (or strings only containing whitespace) in its array will have those empty strings filtered out by the library. For example, if a voice command has the following array values: `[" ", "First", "", "Voice Command"]` will become `["First", "Voice Command"]`.
```

> @![iOS]
> ##### Objective-C
> ```objc
> SDLVoiceCommand *voiceCommand = [[SDLVoiceCommand alloc] initWithVoiceCommands:@[@"\t", @"First", @" ", @"Voice Command"] handler:^{
>     <#Voice command selected#>
> }];
> 
> self.sdlManager.screenManager.voiceCommands = @[voiceCommand];
> ```
> 
> ##### Swift
> ```swift
> let voiceCommand = SDLVoiceCommand(voiceCommands: ["\t", "First", " ", "Voice Command"]) {
>     <#Voice command triggered#>
> }
> 
> sdlManager.screenManager.voiceCommands = [voiceCommand]
> ```
> !@
> 
> @![android, javaSE, javaEE]
> ```java
> // TODO: java code snippets
> ```
> !@
> 
> @![javascript]
> ```js
> // TODO: javascript code snippets
> ```
> !@

Reviewer 2: ```suggestion
```

Not necessary to have code examples

> In a case where all the voice commands defined have only empty strings, the upload request will be aborted and the previous voice commands would not get deleted(e.g. " "). 

Reviewer 2: ```suggestion
If all the voice commands contain only empty strings, the upload request will be aborted and the previous voice commands will remain available. 
```

> More details in the section below.

Reviewer 2: ```suggestion
```
